gammapy/estimators/parameter.py

```diff
@@ -310,24 +380,46 @@ def run(self, datasets, parameter):
         """
         datasets = Datasets(datasets)
         parameter = datasets.parameters[parameter]
+        result = {}
 
         with datasets.parameters.restore_status():
 
             if not self.reoptimize:
                 datasets.parameters.freeze_all()
                 parameter.frozen = False
 
-            result = self.estimate_best_fit(datasets, parameter)
+            result.update(self.estimate_best_fit(datasets, parameter))
+            result["fit_status"] = int(result["success"])
+
             result.update(self.estimate_ts(datasets, parameter))
 
+            # If the fit has not converged, should we compute the other quantities?
+            other_success = True
             if "errn-errp" in self.selection_optional:
                 result.update(self.estimate_errn_errp(datasets, parameter))
+                other_success &= result["success_err"]
 
             if "ul" in self.selection_optional:
                 result.update(self.estimate_ul(datasets, parameter))
+                other_success &= result["success_ul"]
 
             if "scan" in self.selection_optional:
                 result.update(self.estimate_scan(datasets, parameter))
+                other_success &= result["success_scan"]
+
+        if result["fit_status"] == 1 and other_success:
+            result["fit_status"] = 2
```

Comment:
This can go in the loop over AVAILABLE_METHODS if this is given as a dict that link the key to the methods,
as in my previous comment https://github.com/gammapy/gammapy/pull/3518#issuecomment-924066478